core/bloombits/matcher.go

```diff
@@ -95,11 +96,19 @@ func NewMatcher(sectionSize uint64, filters [][][]byte) *Matcher {
 	m.filters = nil
 
 	for _, filter := range filters {
+		// Gather the bit indexes of the filter rule, special casing the nil filter
 		bloomBits := make([]bloomIndexes, len(filter))
 		for i, clause := range filter {
+			if clause == nil {
+				bloomBits = nil
+				break
+			}
 			bloomBits[i] = calcBloomIndexes(clause)
 		}
-		m.filters = append(m.filters, bloomBits)
+		// Accumulate the filter rules if no nil rule was within
+		if bloomBits != nil {
```

Comment:
You could drop this check if you continue-d the outer loop instead of just break-ing out of the inner loop.

Reply:
I don't really like it when multiple nestings are broken out of. Seems a lot harder to follow what's going on, especially since the label can be arbitrarily far away.

Reply:
I kind of agree with both positions, but in this case personally I think it'd be clearer if it did continue the outer loop; the code is compact, so it's clear what the label is referring to. Alternately, you could refactor the inner loop and append into a separate function.